### PR TITLE
Move code generated by `update_lints` to includes

### PR DESCRIPTION
**Pull request by mikerite**
Move code generated by `update_lints` to includes

changelog: none

### CODE REVIEW

**Comment by mikerite** [comment]
The idea with this pull request is to move code generated out of `lib.rs` and into includes.

The benefit is that moves a lot of clutter out of `lib.rs`. It's also easier to do the generation as a whole file can be replaced instead of a region and these files aren't formatted by `rustfmt`.

Some things still be done are:
- [ ] Agree on what the include filenames should be
- [ ] Agree on the format of the generated code (e.g. comment at the top, end in new line etc.)
- [ ] Refactor the `update_lints` code

This is a big change and it might be a bad idea. Feel free to close.

File: clippy_lints/src/lib.deprecated.rs [hunk 1]
```diff
@@ -0,0 +1,70 @@
+// This file was generated by `cargo dev update_lints`.
+// Use that command to update this file and do not edit by hand.
+// Manual edits will be overwritten.
+
+{
```
Comment (flip1995) [inline clippy_lints/src/lib.deprecated.rs:5]: Why is an extra scope necessary here?

File: clippy_lints/src/lib.register_all.rs [hunk 2]
```diff
@@ -0,0 +1,304 @@
+// This file was generated by `cargo dev update_lints`.
+// Use that command to update this file and do not edit by hand.
+// Manual edits will be overwritten.
+
+store.register_group(true, "clippy::all", Some("clippy_all"), vec![
+LintId::of(absurd_extreme_comparisons::ABSURD_EXTREME_COMPARISONS),
```
Comment (flip1995) [inline clippy_lints/src/lib.register_all.rs:6]: I would indent the `LintId::of`.

File: clippy_lints/src/lib.register_lints.rs [hunk 3]
```diff
@@ -0,0 +1,506 @@
+// This file was generated by `cargo dev update_lints`.
+// Use that command to update this file and do not edit by hand.
+// Manual edits will be overwritten.
+
+    store.register_lints(&[
+        #[cfg(feature = "internal-lints")]
```
Comment (flip1995) [inline clippy_lints/src/lib.register_lints.rs:6]: Unnecessary indentation.

**Review by flip1995 — COMMENTED** [review]
I like this change. This will also make syncing with rustc easier, since conflicts in those auto-generated files can just be ignored and `lib.rs` won't have as much churn. 

> Agree on the format of the generated code (e.g. comment at the top, end in new line etc.)

I would add a newline at the end of the file and keep the format consistent. Some files have indentation, some don't. I would format those files as close as possible to what rustfmt would do. (see comments below)

---

Before you continue working on this, I'd wait for more feedback from @rust-lang/clippy though

**Comment by xFrednet** [comment]
I like the idea +1 from me. 

If we continue this, could you please make sure to also include the automatically generated files in our pre-commit hook? :upside_down_face: 

https://github.com/rust-lang/rust-clippy/blob/2316f4da83a3952abca808f1f1b2c7ea0a9ee401/util/etc/pre-commit.sh#L7-L8

**Comment by giraffate** [comment]
Sounds like a good idea, I like it.

File: clippy_lints/src/lib.deprecated.rs (hunk 1 above)
Comment (mikerite) [reply clippy_lints/src/lib.deprecated.rs:5]: Yes. It's required because the include file must parse to a single expression.

**Comment by xFrednet** [comment]
Do we want to move this PR forward once it was rebased? There seems to be no objections so far :upside_down_face:

**Comment by giraffate** [comment]
Sorry for the late reviewing. I've been a bit busy. Please give me a few more days.

**Review by flip1995 — APPROVED** [review]
Generated files LGTM. I didn't review the clippy_dev changes. 

There were no objections to this in the last meeting, so once @giraffate is done with the review, this can be merged.

File: clippy_dev/src/update_lints.rs [hunk 4]
```diff
@@ -150,3 +162,540 @@ pub fn print_lints() {
...
+    let module_names: BTreeSet<_> = lints.map(|l| &l.module).collect();
+
+    let mut output = GENERATED_FILE_COMMENT.to_string();
+    for name in module_names {
+        output.push_str(&format!("mod {};\n", name));
+    }
+    output
+}
+
+/// Generates the list of lint links at the bottom of the README
```
Comment (giraffate) [inline clippy_dev/src/update_lints.rs:269]: nits
```suggestion
/// Generates the list of lint links at the bottom of the CHANGELOG
```

**Review by giraffate — APPROVED** [review]
Sorry for the late review, again.

Over all looks good, Thanks! I made one minor comment.

**Comment by giraffate** [comment]
@bors r+

Thanks!

**Comment by Manishearth** [comment]
Ooh, this has made rustfmt stop working for the entire codebase since the modules are `include!`d and rustfmt doesn't read that.

**Pull request merged**